### PR TITLE
Serialize, Deserialize, and format AsResources, Ipv4Resources and Ipv6Resources

### PR DESCRIPTION
**Pull request by timbru**
Note: this is particularly useful for the Krill project where we need to persist
resources associated with Certificate Authorities.

I introduced the explicit implementations for Ipv4Resources and Ipv6Resources, in addition to IpResources and the (shortlived) IpBlocksForFamily for convenience: it allows for a straightforward Display and FromStr that can be used when (De)Serializing.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,11 +17,12 @@ derive_more     = "^0.14"
 hex             = "^0.3"
 log             = "^0.4"
 openssl         = { version = "^0.10", optional = true }
-quick-xml	= "^0.14"
+quick-xml	    = "^0.14"
 ring            = "^0.13"
-serde		= "^1.0"
+serde   		= "^1.0"
+serde_derive   	= "^1.0"
```
Comment (partim) [inline Cargo.toml]: The serde guide doesn’t include `serde_derive` separately but rather does this:

```
 serde = { version = "1.0", features = ["derive"] }
```

(and then `#[macro_use] extern crate serde` in lib.rs).

File: src/resources/asres.rs [hunk 2]
```diff
@@ -19,6 +19,7 @@ use std::str::FromStr;
 use bcder::{decode, encode};
 use bcder::Tag;
 use bcder::encode::PrimitiveContent;
+use serde::{de, Deserialize, Deserializer, Serialize, Serializer};
```
Comment (partim) [inline src/resources/asres.rs]: Matter of style: I keep module imports and type imports on separate lines, i.e.,

```
use serde::de;
use serde::{Deserialize, ...};
```

File: src/resources/asres.rs [hunk 3]
```diff
@@ -306,14 +337,54 @@ impl FromStr for AsBlocks {
...
+                let block = AsBlock::from_str(&el)?;
+                builder.push(block);
+            }
         }
 
         Ok(builder.finalize())
     }
 }
 
+//--- PartialEq
```
Comment (partim) [inline src/resources/asres.rs]: If you implement `PartialEq` and `Eq` for `resources::chain::SharedChain` (and all the other chains, while you are at it), you can simply derive them here.

I suspect that you can even just derive them for the chains, given that they are ordered and non-continuous (i.e., the same set of resources should result in the exact same chain).

File: src/resources/asres.rs [hunk 4]
```diff
@@ -306,14 +337,54 @@ impl FromStr for AsBlocks {
...
+        }
+        true
+    }
+}
+
+impl Eq for AsBlocks {}
+
+//--- Serialize
+
+impl Serialize for AsBlocks {
```
Comment (partim) [inline src/resources/asres.rs]: Unless you totally want them to be serialized in their string representation, you can probably derive {Des,S}erialize all the way down to AsBlock/IpBlock?

Comment (timbru) [reply src/resources/asres.rs]: I like these to be readable (well a json value of readable) for debugging and use in configuration files.

File: Cargo.toml (hunk 1 above)
Comment (timbru) [reply Cargo.toml]: pushed

File: src/resources/asres.rs (hunk 2 above)
Comment (timbru) [reply src/resources/asres.rs]: pushed

File: src/resources/asres.rs (hunk 3 above)
Comment (timbru) [reply src/resources/asres.rs]: pushed.. implemented for chains - did not try deriving there yet, Block does not implement Eq, although its Item does. Not sure.. but no time right now - have to be in a call.

**Pull request merged**